src/httpserver.cpp

```diff
@@ -480,6 +475,14 @@ void StopHTTPServer()
         workQueue->WaitExit();
         delete workQueue;
     }
+    if (eventBase) {
+        LogPrint("http", "Waiting for HTTP event thread to exit\n");
+        // Give event loop a few seconds to exit (to send back last RPC responses), then break it
+        if (!threadHTTP.try_join_for(boost::chrono::milliseconds(2000))) {
```

Comment:
> @MarcoFalke but this doesn't stop new connections (or is that done elsewhere?) in that waiting time.  So conceptually, the status quo could be the same?

It does:
```
        // Unlisten sockets
        BOOST_FOREACH (evhttp_bound_socket *socket, boundSockets) {
            evhttp_del_accept_socket(eventHTTP, socket);
        }
        // Reject requests on current connections
        evhttp_set_gencb(eventHTTP, http_reject_request_cb, NULL);
```
This closes the port and makes new requests error (so that no new work items are created). But that's best-effort. Current connections could still do e.g. a slowloris attack. The two seconds are a belt and suspenders, nothing more.